src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java

```diff
@@ -54,57 +43,11 @@ public ProviderSqlSource(Configuration configuration, Object provider) {
    * @since 3.4.5
    */
   public ProviderSqlSource(Configuration configuration, Object provider, Class<?> mapperType, Method mapperMethod) {
-    String providerMethodName;
-    try {
-      this.configuration = configuration;
-      this.sqlSourceParser = new SqlSourceBuilder(configuration);
-      this.providerType = (Class<?>) provider.getClass().getMethod("type").invoke(provider);
-      providerMethodName = (String) provider.getClass().getMethod("method").invoke(provider);
-
-      for (Method m : this.providerType.getMethods()) {
-        if (providerMethodName.equals(m.getName())) {
-          if (m.getReturnType() == String.class) {
-            if (providerMethod != null){
-              throw new BuilderException("Error creating SqlSource for SqlProvider. Method '"
```

Comment:
Oh, I see. This was not your code. This originally was there. My comments still apply

Reply:
This abstract class is extracted from [ProviderSqlSource](https://github.com/mybatis/mybatis-3/blob/master/src/main/java/org/apache/ibatis/builder/annotation/ProviderSqlSource.java), and this part of the code is not modified.